### PR TITLE
[device]

### PR DESCRIPTION
**Pull request by proyan**
* Change naming convention of CONTROL_TYPE.
* Add function setNoIntegration in the SoTAbstract External Interface
* Move robotState signal from sot-hrp2-device to sot-core.
* Create additional signal robotVelocity to map the velocity sensor data from robot/simulators

Deals with stack-of-tasks/sot-core#39
Deals with stack-of-tasks/sot-torque-control#3

TODO: Change robotState and robotVelocity signal names.
TODO: Change name of state signal

### CODE REVIEW

**Comment by proyan** [comment]
@andreadelprete This changes the CONTROL_TYPE strings, so it might affect some of your scripts.

File: include/sot/core/device.hh [hunk 1]
```diff
@@ -40,15 +40,15 @@ namespace dynamicgraph {
     /// Define the type of input expected by the robot
     enum ControlInput
     {
-      CONTROL_INPUT_POSITION=0,
-      CONTROL_INPUT_VELOCITY=1,
-      CONTROL_INPUT_ACCELERATION=2,
+      CONTROL_INPUT_NO_INTEGRATION=0,
+      CONTROL_INPUT_ONE_INTEGRATION=1,
```
Comment (olivier-stasse) [inline include/sot/core/device.hh]: I do not understand why your changing the naming here. 
I think this is more confusing.
If you need to have no integration you should add one more possibilty

Comment (olivier-stasse) [inline include/sot/core/device.hh]: Do you mean that you are integrating one time or two times ?
Integrating two times some of the quantities would be meaningless (for instance velocity)

Comment (proyan) [inline include/sot/core/device.hh]: Double integration is meant for the acceleration mode.

Comment (proyan) [inline include/sot/core/device.hh]: > If you need to have no integration you should add one more possibilty

Currently we don't have an integration in the position control type. However, this is misleading when used in the torque control mode, which also requires no control integration

File: include/sot/core/device.hh [hunk 2]
```diff
@@ -93,6 +93,7 @@ namespace dynamicgraph {
       void setVelocitySize(const unsigned int& size);
       virtual void setVelocity(const dg::Vector & vel);
       virtual void setSecondOrderIntegration();
+      virtual void setNoIntegration();
```
Comment (andreadelprete) [inline include/sot/core/device.hh]: The method setNoIntegration seems redundant with respect to the method setControlInputType, which can be used to specify the control input type CONTROL_INPUT_NO_INTEGRATION

Comment (proyan) [inline include/sot/core/device.hh]: enum ControlInput is a member of the Device class. 
setSecondOrderIntegration and setNoIntegration are needed in the AbstractSotExternalInterface class, which is independent from the Device and defines the API for simulators and robots.

I don't think using an enum whose values are defined in a different class is a good idea for the Abstract API

File: include/sot/core/device.hh (hunk 1 above)
Comment (olivier-stasse) [inline include/sot/core/device.hh]: Theoretically a direct dynamic algorithm like CRBA would provide you position based on torque.

Comment (proyan) [inline include/sot/core/device.hh]: The use of ControlInput in the device class is in the increment and integrate functions to get the state and velocity signals from the input control signal. The CONTROL_TYPE_POSITION is the only type where the velocity signal is not being filled from the control signal or its integral.

For the torque control mode, we need to not integrate the input control signal. We are using the name CONTROL_INPUT_POSITION for that, which is misleading.

Comment (proyan) [inline include/sot/core/device.hh]: > Theoretically a direct dynamic algorithm like CRBA would provide you position based on torque.
Do you mean ABA?
I don't think we need to implement the full integration scheme for torque control, unless we are implementing a simulation device in SoT (which I think Andrea did for device control). In the device class, the requirement would be just to populate the outgoing signals.

File: include/sot/core/device.hh (hunk 2 above)
Comment (andreadelprete) [inline include/sot/core/device.hh]: if `setNoIntegration` is only needed in the `AbstractSotExternalInterface`, then why did you add it to `Device`?
